src/util/rocksdb.rs

```diff
@@ -47,6 +50,48 @@ pub fn new_engine(path: &str, cfs: &[&str]) -> Result<DB, String> {
     new_engine_opt(opts, path, cfs, cfs_opts)
 }
 
+fn check_column_families(path: &str, used_cfs: &[&str]) -> Result<(), String> {
+    if !db_exist(path) {
+        return Ok(());
+    }
+
+    // List all column families in current db.
+    let mut opts = Options::new();
+    opts.create_if_missing(false);
+    let cfs_list = match DB::list_column_families(&opts, path) {
+        Ok(cfs) => cfs,
+        Err(e) => return Err(e),
+    };
+
+    let existed: HashSet<&str> = cfs_list.iter().map(|v| v.as_str()).collect();
+    let needed: HashSet<&str> = used_cfs.iter().map(|v| *v).collect();
+
+    // Open db.
+    let mut cfs_opts: Vec<Options> = vec![];
+    for _ in 0..cfs_list.len() {
+        cfs_opts.push(Options::new());
+    }
+    let cfs_ref_opts: Vec<&Options> = cfs_opts.iter().collect();
+    let cfs_list_str: Vec<&str> = cfs_list.iter().map(|cf| cf.as_str()).collect();
+    let mut db = DB::open_cf(&opts, path, &cfs_list_str, &cfs_ref_opts).unwrap();
```

Comment:
Can it cause data lose? For example, we may using different recovery mode than expected to open db.